src/core/sock.c

```diff
@@ -122,6 +122,7 @@ int nn_sock_init (struct nn_sock *self, struct nn_socktype *socktype, int fd)
     self->linger = 1000;
     self->sndbuf = 128 * 1024;
     self->rcvbuf = 128 * 1024;
+    self->rcvmaxsize = 512 * 1024 * 1024;
```

Comment:
Agree. When seeing this feature, this is the first priority I considered -- setting a decently robust/secure default threshold. 1MB feels acceptable, but it could be argued up to ~10MB or so. 512MB is too high a default today in 2015 :-)